app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -72,23 +70,55 @@ class DatabaseCleanupManagerImpl(
         launch {
             val configuration = configurationRepository.observeConfiguration().blockingFirst()
 
-            val thresholdRedMessages = ZonedDateTime.now().minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Red, thresholdRedMessages)
-
-            val thresholdYellowMessages = ZonedDateTime.now().minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Yellow, thresholdYellowMessages)
-
-            val thresholdGreenMessages = ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.Revoke.Suspicion, thresholdGreenMessages)
+            val thresholdRedMessages = ZonedDateTime.now()
+                .minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.InfectionLevel.Red,
+                thresholdRedMessages
+            )
+
+            val thresholdYellowMessages = ZonedDateTime.now()
```

Comment:
I believe we must be careful with "hours" here

Image if it´s today 18:00 and we want to delete the keys of the last 2 days -> 2*24+18 = 66 hours of deletion
Image if it´s today 01:00 and we want to delete the keys of the last 2 days -> 2*24+1 = 49 hours of deletion

The framwork will introduce TEKs that are not 24 hour compatible, our code should be compatible. 

The configuration will be in days if I remember the discussion correctly with @uluckas

Reply:
use the startOfTheDay() maybe!

Reply:
Very good point 💡

Reply:
This line is part of the `cleanupReceivedInfectionMessages` which cleans up the downloaded messages, the download functionality has not been touched in this MR. 

The logic that takes care of cleaning up the sent temporary keys is in `cleanupSentTemporaryExposureKeys` and the difference is that it uses rollingStartIntervalNumber ( = unix timestamp / 600) which is less precise than a unix timestamp. I would say the rollingStartIntervalNumber should have a granularity of ~2 hours.

I was not sure exactly what is the best approach in deleting the old keys, from my understanding is possible to have multiple rollingStartInterval per day. 

Please check this method and let's sync on what is the best approach to delete them

Reply:
@mihbt Please remove all keys older than 2 days of time 00:00 of the day.